### ISSUE TITLE
crash on startup with nosave

### ISSUE THREAD

**Issue by tg-x** [atlas-engineer/nyxt#1262]
**Describe the bug**

Nyxt crashes on startup when launched with the nosave profile.

**Precise recipe to reproduce the issue**

1. started with `nyxt -d nosave` a couple of times in a row would always lead to a crash upon loading the history.
2. started it as `nyxt` without the nosave argument, then it loaded fine
3. after this nyxt start again with the nosave profile without crash

perhaps related to a specific state saved in the history that was later fixed by using the default profile?
I can't reproduce this anymore after this, so not sure how to debug, but perhaps the log below helps.

**Information**

- OS name+version: NixOS running the pre-built nyxt-2-pre-6 guix pack I downloaded.
- Desktop environment / Window manager name+version: wayland + sway

 **Output when started from a shell**

```
> nyxt -d nosave

(process:3512): Gtk-WARNING **: 19:51:22.433: Locale not supported by C library.
        Using the fallback 'C' locale.
<INFO> [19:51:22] Listening to socket "/home/dev/.local/share/nyxt/nyxt.socket".
Nyxt version 2-pre-release-6

(process:3538): Gtk-WARNING **: 19:51:22.975: Locale not supported by C library.
        Using the fallback 'C' locale.

(process:3550): Gtk-WARNING **: 19:51:23.037: Locale not supported by C library.
        Using the fallback 'C' locale.

(process:3590): Gtk-WARNING **: 19:51:23.163: Locale not supported by C library.
        Using the fallback 'C' locale.

(process:3628): Gtk-WARNING **: 19:51:23.332: Locale not supported by C library.
        Using the fallback 'C' locale.
<INFO> [19:51:24] Loading history of 11 URLs from "/home/dev/.local/share/nyxt/history/default.lisp".
<INFO> [19:51:24] Loading history of 11 URLs from "/home/dev/.local/share/nyxt/history/default.lisp".
<INFO> [19:51:24] Loading history of 11 URLs from "/home/dev/.local/share/nyxt/history/default.lisp".

(process:3666): Gtk-WARNING **: 19:51:24.298: Locale not supported by C library.
        Using the fallback 'C' locale.
<INFO> [19:51:24] Loading "https://nyxt.atlas.engineer/start".
Unhandled SIMPLE-ERROR in thread #<SB-THREAD:THREAD "cl-cffi-gtk main thread" RUNNING
                                    {10078A6973}>:
  Owner with identifier "4" does not exist

Backtrace for: #<SB-THREAD:THREAD "cl-cffi-gtk main thread" RUNNING {10078A6973}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<SIMPLE-ERROR "Owner with identifier ~s does not exist" {100E13C273}> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK SB-EXT:*INVOKE-DEBUGGER-HOOK* #<SIMPLE-ERROR "Owner with identifier ~s does not exist" {100E13C273}>)
2: (INVOKE-DEBUGGER #<SIMPLE-ERROR "Owner with identifier ~s does not exist" {100E13C273}>)
3: (ERROR "Owner with identifier ~s does not exist" "4")
4: (HISTORY-TREE:SET-CURRENT-OWNER #<HISTORY-TREE:HISTORY-TREE {100D3D4043}> "4")
5: ((FLET "CLEANUP-FUN-8" :IN NYXT::HISTORY-ADD)) [cleanup]
6: ((FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN NYXT::HISTORY-ADD))
7: ((FLET "WITHOUT-INTERRUPTS-BODY-11" :IN SB-THREAD::CALL-WITH-RECURSIVE-LOCK))
8: (SB-THREAD::CALL-WITH-RECURSIVE-LOCK #<FUNCTION (FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN NYXT::HISTORY-ADD) {7F0A9E95D84B}> #<SB-THREAD:MUTEX "Anonymous recursive lock" owner: #<SB-THREAD:THREAD "cl-cffi-gtk main thread" RUNNING {10078A6973}>> T NIL)
9: (NYXT::HISTORY-ADD #<QURI.URI.HTTP:URI-HTTPS https://#1=nyxt.atlas.engineer/start> :TITLE "Nyxt browser" :BUFFER #<USER-WEB-BUFFER {100D4597F3}>)
10: ((:METHOD ON-SIGNAL-NOTIFY-URI (NYXT/WEB-MODE:WEB-MODE T)) #<NYXT/WEB-MODE:USER-WEB-MODE {100D921813}> #<QURI.URI.HTTP:URI-HTTPS https://#1=nyxt.atlas.engineer/start>) [fast-method]
11: ((:METHOD ON-SIGNAL-NOTIFY-TITLE (NYXT/WEB-MODE:WEB-MODE T)) #<NYXT/WEB-MODE:USER-WEB-MODE {100D921813}> #<QURI.URI.HTTP:URI-HTTPS https://#1=nyxt.atlas.engineer/start>) [fast-method]
12: ((:METHOD ON-SIGNAL-NOTIFY-TITLE (BUFFER T)) #<USER-WEB-BUFFER {100D4597F3}> #<unused argument>) [fast-method]
13: (GOBJECT::CALL-WITH-RESTARTS #<FUNCTION (LAMBDA (NYXT::WEB-VIEW NYXT::PARAM-SPEC) :IN FFI-BUFFER-MAKE) {100D7F213B}> (#<CL-WEBKIT2:WEBKIT-WEB-VIEW {100D7F0893}> #<PROPERTY gchararray #<GTYPE :name "WebKitWebView" :id 35956944> . title (flags: readable)>))
14: ((LAMBDA (GOBJECT::CLOSURE GOBJECT::RETURN-VALUE GOBJECT::COUNT-OF-ARGS GOBJECT::ARGS GOBJECT::INVOCATION-HINT GOBJECT::MARSHAL-DATA) :IN "/gnu/store/5xsc8k03vwv99z6v06dkiryvbz7hsxf6-sbcl-cl-cffi-gtk-0.11.2-2.e9a46df/share/common-lisp/sbcl/cl-cffi-gtk/gobject/gobject.si
gnals.lisp") #.(SB-SYS:INT-SAP #X7F0A8C282580) #.(SB-SYS:INT-SAP #X00000000) 2 #.(SB-SYS:INT-SAP #X7F0A9E95DF00) #<unused argument> #<unused argument>)
15: ((LAMBDA (SB-ALIEN::ARGS-POINTER SB-ALIEN::RESULT-POINTER FUNCTION) :IN "/gnu/store/5xsc8k03vwv99z6v06dkiryvbz7hsxf6-sbcl-cl-cffi-gtk-0.11.2-2.e9a46df/share/common-lisp/sbcl/cl-cffi-gtk/gobject/gobject.signals.lisp") #<unavailable argument> #<unavailable argument> #<una
vailable argument>)
16: ("foreign function: funcall_alien_callback")
17: ("foreign function: callback_wrapper_trampoline")
18: ("foreign function: #x5010086B")
19: ((LAMBDA NIL :IN GTK:ENSURE-GTK-MAIN))
20: ((LAMBDA NIL :IN BORDEAUX-THREADS::BINDING-DEFAULT-SPECIALS))
21: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
22: ((FLET "WITHOUT-INTERRUPTS-BODY-11" :IN SB-THREAD::RUN))
23: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
24: ((FLET "WITHOUT-INTERRUPTS-BODY-4" :IN SB-THREAD::RUN))
25: (SB-THREAD::RUN)
26: ("foreign function: call_into_lisp")
27: ("foreign function: funcall1")

unhandled condition in --disable-debugger mode, quitting
```

**Comment by Ambrevar** [comment]
This crash should be mitigated with 1cb518ed921aaa5bd40eaaf89fd1531c01e9870e.  I'm not sure why the condition is raised in the first place though.

**Comment by Ambrevar** [comment]
More "missing owner" bugs are fixed with eb418c140b12defc0617b90862cf2b3a0769673b.
